### PR TITLE
Add 'ping' command

### PR DESCRIPTION
**Pull request by jsullivan3**
To perform memberlist pings as a network diagnostic tool, add 'ping' CLI command, plumb through to the client and server, and then invoke the Memberlist.Ping() method.

### CODE REVIEW

File: command/agent/agent.go [hunk 1]
```diff
@@ -1473,3 +1473,28 @@ func (a *Agent) SerfQuery(name string, payload []byte, params *serf.QueryParam)
...
+	if err != nil {
+		a.logger.Printf("[WARN] agent: failed to start user serf ping: %v", err)
+		return nil, err
+	}
+
+	resp := serfPingResponse{
+		Success: cresp.Success,
+		RTT:     cresp.RTT,
+	}
+	return &resp, err
```
Comment (ryanbreen) [inline command/agent/agent.go]: Is the idiomatic thing in this case to return `&resp, nil` instead?

File: command/agent/rpc.go [hunk 2]
```diff
@@ -616,6 +626,25 @@ func (i *AgentRPC) handleSerfQuery(client *rpcClient, seq uint64) error {
...
+func (i *AgentRPC) handleSerfPing(client *rpcClient, seq uint64) error {
+	var req serfPingRequest
+	if err := client.dec.Decode(&req); err != nil {
+		return fmt.Errorf("decode failed: %v", err)
+	}
+
+	i.logger.Printf("[DEBUG] agent.rpc: handling serf ping request")
+
+	// Start the serf query
+	resp, err := i.agent.SerfPing(req.Name)
```
Comment (ryanbreen) [inline command/agent/rpc.go:638]: Shouldn't we check for err and return here?

**Comment by ryanbreen** [comment]
:+1: 

This looks good to me.  Just a couple of stylistic questions.

File: command/agent/rpc.go (hunk 2 above)
Comment (jsullivan3) [inline command/agent/rpc.go:638]: Excellent point.  If the error gets returned here, it does not get propagated to the client.  An error such as the node not being in the member list would not be propagated.

I think I'd prefer something similar to the following at line 645:

```
if replyErr := client.Send(&header, &resp); replyErr != nil {
    return replyErr
else
    return err
fi
```

Would you be comfortable with this change?

Comment (jsullivan3) [inline command/agent/rpc.go:638]: Looking closer at the code, the existing pattern can also be found in handleForceLeave() and handleJoin().  The suggestion I had for line 645 cannot be found elsewhere.  I think with this observation I would prefer to not modify the code so as to follow the precedent set by other implementations.

Comment (ryanbreen) [inline command/agent/rpc.go:638]: Does it make sense to proceed from line 638 if there's an error setting up the serf ping query?  I was thinking we should just do an `if err return err` here.

File: command/agent/agent.go (hunk 1 above)
Comment (jsullivan3) [inline command/agent/agent.go]: This suggestion would make the code more consistent with other functions in this file.  I'll make this change.  Thanks!

File: command/agent/rpc.go (hunk 2 above)
Comment (ryanbreen) [inline command/agent/rpc.go:638]: Cool.  Adhering to the existing patterns makes sense.

**Pull request merged**